src/mon/FSCommands.cc

```diff
@@ -607,6 +607,27 @@ class SetHandler : public FileSystemCommandHandler
         return r;
       }
 
+      if (!allow) {
+        if (!mon->osdmon()->is_writeable()) {
+          // not allowed to write yet, so retry when we can
+          mon->osdmon()->wait_for_writeable(op, new PaxosService::C_RetryMessage(mon->mdsmon(), op));
+          return -EAGAIN;
+        }
+        std::vector<mds_gid_t> to_fail;
+        for (const auto& [gid, info]: fs->mds_map.get_mds_info()) {
+          if (info.state == MDSMap::STATE_STANDBY_REPLAY) {
+            to_fail.push_back(gid);
```

Comment:
Why is the vector to_fail necessary? Why not fail the mds inside this for loop?

Reply:
Because the MDS would be removed from the map returned by `fs->mds_map.get_mds_info()` which would disrupt iterating through the map.

Reply:
Ahh got it!